lms/services/grouping.py

```diff
@@ -21,10 +21,10 @@ def generate_authority_provided_id(
         type_: Grouping.Type,
     ):
         if type_ == Grouping.Type.COURSE:
+            assert parent is None, "Course groupings can't have a parent"
```

Comment:
I added strings to the exceptions. This means that you see ``AssertionError: Course groupings can't have a parent` in the traceback instead of just `AssertionError` and the tests can use `match` to make sure they're catching the `AssertionError` that they expected and not another one.